ksql-engine/src/test/java/io/confluent/ksql/function/udf/KudfTester.java

```diff
@@ -0,0 +1,368 @@
+/*
+ * Copyright 2018 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.confluent.ksql.function.udf;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+import io.confluent.ksql.util.KsqlException;
+
+import static junit.framework.TestCase.fail;
+
+/**
+ * Helper for testing KUDFs.
+ */
+public class KudfTester {
+
+  private static final int NO_MAX_ARGS = Integer.MAX_VALUE;
+
+  private static final Map<Class<?>, Object>
+      DEFAULT_INSTANCES =
+      ImmutableMap.<Class<?>, Object>builder()
+          .put(String.class, "Hello")
+          .put(Boolean.class, true)
+          .put(Short.class, (short) 22)
+          .put(Integer.class, 66L)
+          .put(Long.class, 42L)
+          .put(Double.class, 12.345)
+          .put(Float.class, 567.89f)
```

Comment:
How about array/map and struct arguments?

Reply:
Can be added when needed. (I deliberately kept it simple initially)